packages/babel-plugin-transform-nullish-coalescing-operator/src/index.js

```diff
@@ -0,0 +1,31 @@
+import syntaxNullishCoalescingOperator from "babel-plugin-syntax-nullish-coalescing-operator";
+
+export default function({ types: t }) {
+  return {
+    inherits: syntaxNullishCoalescingOperator,
+
+    visitor: {
+      LogicalExpression(path) {
+        const { node } = path;
+        if (node.operator !== "??") {
+          return;
+        }
+
+        const scope = path.scope.parent || path.scope;
+        const ref = scope.generateUidIdentifierBasedOnNode(node.left);
+        scope.push({ id: ref });
+
+        path.replaceWith(
+          t.sequenceExpression([
+            t.assignmentExpression("=", ref, node.left),
+            t.conditionalExpression(
+              t.binaryExpression("!=", ref, t.nullLiteral()),
```

Comment:
`ref` needs to be cloned here.

Reply:
What's the rule for when a node needs to be cloned? When it is passed to multiple builders?